test/renaissance_web/controllers/auction_controller_test.exs

```diff
@@ -29,86 +28,84 @@ defmodule RenaissanceWeb.AuctionControllerTest do
+    test "POST /auctions rejects auction when end date is in the past", %{conn: conn} do
       invalid_datetime = %{@valid_end | year: "1999"}
       invalid_params = %{@auction_one_params | end_auction_at: invalid_datetime}
       conn = post(conn, "/auctions", invalid_params)
 
       assert html_response(conn, 200) =~ "in the future"
     end
 
-    test "POST /auctions fails to create auction when the price is 0 dollars", %{conn: conn} do
-      invalid_params = %{@auction_one_params | price: "0.00"}
-      conn = post(conn, "/auctions", invalid_params)
-
-      assert html_response(conn, 200) =~ "must be greater than 0"
-    end
-
     test "GET /auctions displays all auctions when signed in", %{conn: conn} do
-      auction_two_params = %{
-        title: "Test Two Title",
-        description: "Test two description.",
-        end_auction_at: %{@valid_end | day: @valid_end.day + 1},
-        price: "15.00"
-      }
-
       conn =
         conn
         |> post("/auctions", @auction_one_params)
-        |> post("/auctions", auction_two_params)
+        |> post("/auctions", @auction_two_params)
         |> get("/auctions")
 
       assert html_response(conn, 200) =~ @auction_one_params.title
       assert html_response(conn, 200) =~ "$" <> @auction_one_params.price
       assert html_response(conn, 200) =~ @auction_one_params.description
 
-      assert html_response(conn, 200) =~ auction_two_params.title
-      assert html_response(conn, 200) =~ "$" <> auction_two_params.price
-      assert html_response(conn, 200) =~ auction_two_params.description
+      assert html_response(conn, 200) =~ @auction_two_params.title
+      assert html_response(conn, 200) =~ "$" <> @auction_two_params.price
+      assert html_response(conn, 200) =~ @auction_two_params.description
+
+      refute html_response(conn, 200) =~ ~s(class="countdown")
+    end
+
+    test "GET /auctions/:id returns only the specified auction", %{conn: conn} do
+      conn =
+        conn
+        |> post("/auctions", @auction_one_params)
+        |> post("/auctions", @auction_two_params)
+
+      auction_one_id = Repo.get_by(Auction, title: @auction_one_params.title).id
+      conn = get(conn, "/auctions/#{auction_one_id}")
+
+      assert html_response(conn, 200) =~ @auction_one_params.title
+      refute html_response(conn, 200) =~ @auction_two_params.title
+    end
+
+    test "GET /auctions/:id returns more auction details than index", %{conn: conn} do
+      conn = conn |> post("/auctions", @auction_one_params)
+
+      auction_one_id = Repo.get_by(Auction, title: @auction_one_params.title).id
+      conn = get(conn, "/auctions/#{auction_one_id}")
+
+      assert html_response(conn, 200) =~ ~s(class="countdown")
+      assert html_response(conn, 200) =~ "Auction ends in"
+      assert html_response(conn, 200) =~ " years"
+    end
+
+    test "GET /auctions/:id fails when invalid id", %{conn: conn} do
+      assert_error_sent 404, fn ->
+        get(conn, "/auctions/285")
```

Comment:
I usually use 0 to test 404 in case other records exist and what not since 0 should never be a primary key

Reply:
it makes it more obvious it shouldn't work

Reply:
Fair enough–the table should be empty there, but I realize that's not clear.